pkg/segment/writer/startree_test.go

```diff
@@ -817,25 +892,52 @@ func TestStarTree2(t *testing.T) {
 		ss.wipBlock.blockSummary.RecCount += 1
 	}
 
-	groupByCols := []string{"brand", "color", "type"}
-	mColNames := []string{"price", "perf", "rating"}
-
-	gcWorkBuf := make([][]string, len(groupByCols))
-	for colNum := 0; colNum < len(groupByCols); colNum++ {
-		gcWorkBuf[colNum] = make([]string, MaxAgileTreeNodeCount)
-	}
-
 	// basic test
 	var builder StarTreeBuilder
-	for trial := 0; trial < 10; trial += 1 {
+	for trial := 0; trial < 1; trial += 1 {
 		builder.ResetSegTree(groupByCols, mColNames, gcWorkBuf)
 		err := builder.ComputeStarTree(&ss.wipBlock)
 		assert.NoError(t, err)
 		root := builder.tree.Root
 
+		encMap, err := builder.DeriveEnc(ss.wipBlock, allGrpVals)
+		assert.NoError(t, err)
+
+		tree := make(map[string][]utils.CValueEnclosure)
+		getTree(t, root, tree, "")
+
 		_, err = builder.EncodeStarTree(ss.SegmentKey)
 		assert.NoError(t, err)
 
+		// check the tree structure
+		assert.Equal(t, 13, len(tree))
+		key := createKey(root, encMap, groupByCols, []string{"toyota", "green", "sedan"})
+		testAggs(t, []int{10, 10, 10, 1, 9, 9, 9, 1, 5, 5, 5, 1}, tree[key])
+		key = createKey(root, encMap, groupByCols, []string{"toyota", "yellow", "sedan"})
+		testAggs(t, []int{8, 8, 8, 1, 7, 7, 7, 1, 3, 3, 3, 1}, tree[key])
+		key = createKey(root, encMap, groupByCols, []string{"toyota", "red", "suv"})
```

Comment:
This is hard for me to verify.

It might be good to have this for one of your tests, to make sure the tree structure is exactly what we expect. But for most of your tests, it might be better to do something like this:
1. Ingest some data
2. Run various queries on it (not necessarily a full SPL query; just get the relevant data out of the tree and aggregated and/or bucketed correctly) and ensure they all give the correct results (it should be fairly easy for someone to read the code and verify they're the expected results)
3. Drop a column (or multiple columns)
4. Run the same queries as before that you still have all the columns for, and check that they give the same results

Reply:
I have verified this using an end-to-end test procedure mentioned above. Creating an end-to-end unit test will take time because a lot of setup is necessary.